builder/dockerfile/internals.go

```diff
@@ -248,7 +248,6 @@ func (b *Builder) download(srcURL string) (fi builder.FileInfo, err error) {
 	progressReader := progress.NewProgressReader(resp.Body, progressOutput, resp.ContentLength, "", "Downloading")
 	// Download and dump result to tmp file
 	if _, err = io.Copy(tmpFile, progressReader); err != nil {
-		tmpFile.Close()
```

Comment:
I consider maintainability an advantage. :wink: But I don't really mind all that much.